### PR TITLE
Adding CI with AppVeyor

### PR DESCRIPTION
**Pull request by alvpickmans**
Started with issue #5 

**Changes**:
- Update of `SpeckleDynamo.csproj` to also generate a `SpeckleDynamo` folder on the target directory to easily create a zip file artifact from it.

**Additions**
- appveyor.yml file

@didimitrie @fraguada @teocomi  If this is ok, could you add a project for SpeckleDynamo on the AppVeyor SpeckleWorks' account?

### CODE REVIEW

**Comment by didimitrie** [comment]
Sure, actually this is great! Do you have an appveyor account? If so there should be a way to get you in on it.

**Comment by alvpickmans** [comment]
I do since yesterday!  It is my personal email aortegapickmans@gmail.com

**Comment by didimitrie** [comment]
@alvpickmans , done: https://ci.appveyor.com/project/SpeckleWorks/speckledynamo
I've added you as a collaborator too to the whole speckleworks appveyor gig!

**Comment by pearswj** [comment]
Hey @alvpickmans, I have a few thoughts from setting this up for other Speckle repos:

* You can copy the Slack notifications from [SpeckleRhino appveyor.yml](https://github.com/speckleworks/SpeckleRhino/blob/master/appveyor.yml#L33)
* What's the relationship between the build "version" here, the `AssemblyVersion` attributes and the version number in `pkg.json`?
* I think there's a bunch of duplication between `AfterBuildDebug` and `AfterBuildRelease` that can be tidied up </nitpicking>

@didimitrie We should get CI turned on before this PR gets merged!

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+before_build:
+  - nuget restore
+
+build:
+  project: SpeckleDynamo.sln
+  verbosity: minimal
+
+after_build:
+  # zip everything into a zip file containing package folder structure
+  - 7z a SpeckleDynamo-$(appveyor_build_version).zip bin\$(configuration)\$(ProjectName)
```
Comment (pearswj) [inline appveyor.yml]: You could try...
```
7z a -r SpeckleDynamo-$(appveyor_build_version).zip %APPVEYOR_BUILD_FOLDER%\SpeckleDynamo\bin\Release\*
```
**EDIT:** I figured out where the `bin/` `extra/` and `pkg.json` folders/files were being created...
```
7z a -r SpeckleDynamo-$(appveyor_build_version).zip %APPVEYOR_BUILD_FOLDER%\SpeckleDynamo\bin\Release\SpeckleDynamo\*
```

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# General Configuration
+version: 0.0.{build}
```
Comment (pearswj) [inline appveyor.yml]: What's the relationship between the build "version" here, the AssemblyVersion attributes and the version number in pkg.json?

**Comment by alvpickmans** [comment]
Thanks for the feedback @pearswj 

- I didn't want to set that up just yet until testing that AppVeyor works fine and creates the artifacts properly, so to not fill Slack with failed checks.
- There's none to be honest. The way Dynamo handles packages is a bit cumbersome and its Package Manager has been questioned a lot recently. I set up the yml file so the AppVeyor version at least matches the major and minor version fields on the `pkg.json`. I've just started with CI, but I think the `AssemblyVersion` can get updated with the AppVeyor one once all check have passed?
- Totally agree, it was a quick and dirty fix to have this rolling but it also seemed odd to me. I will have a check later on.

File: appveyor.yml (hunk 2 above)
Comment (pearswj) [reply appveyor.yml]: @alvpickmans AssemblyInfo patching could work here for sure!

Comment (pearswj) [reply appveyor.yml]: On the other projects I've opted to scrap the `0.0.{build}` version number in favour of a simple `{build}`. It all depends where you want the single source of truth to be 🤷🏻‍♂️

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+before_build:
+  - nuget restore
+
+build:
+  project: SpeckleDynamo.sln
+  verbosity: minimal
+
+after_build:
+  # zip everything into a zip file containing package folder structure
+  - 7z a SpeckleDynamo-$(appveyor_build_version).zip %APPVEYOR_BUILD_FOLDER%\SpeckleDynamo\bin\Release\SpeckleDynamo\*
```
Comment (pearswj) [inline appveyor.yml]: @alvpickmans Just spotted this... `$(appveyor_build_version)` should be `%APPVEYOR_BUILD_VERSION%`

Comment (pearswj) [reply appveyor.yml]: I think this will fix the missing build artifacts

Comment (alvpickmans) [reply appveyor.yml]: you areprobably right, I will update this now and if fails I will just leave it as SpeckleDynamo.zip :)

**Comment by pearswj** [comment]
LGTM! (Or maybe that should be "Let's Get This [Squashed](https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits)"! :) )

**Comment by alvpickmans** [comment]
just a final polishing first!

**Pull request merged**